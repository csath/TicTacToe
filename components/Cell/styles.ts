import { StyleSheet } from 'react-native';
import Layout from '../../constants/Layout';

const styles = StyleSheet.create({
  container: {
    height: (Layout.window.width - 60) / 3,
    width: (Layout.window.width - 60) / 3,
    justifyContent: "center",
    alignItems: "center",
  },
});

export default styles;
