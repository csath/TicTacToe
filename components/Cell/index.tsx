import React from "react";
import { Pressable, PressableProps, View } from "react-native";
import Ionicons from "@expo/vector-icons/Ionicons";
import { CellOccupancy } from "../../types";
import Colors from "../../constants/Colors";
import styles from "./styles";

type CellProps = {
  player: CellOccupancy;
  wrapperStyle?: View["props"]["style"];
} & PressableProps;

const Cell = ({ player, wrapperStyle, ...props }: CellProps) => {
  const getIcon = (player: CellOccupancy) => {
    switch (player) {
      case CellOccupancy.PLAYER_ONE:
        return (
          <Ionicons
            name="md-radio-button-off"
            size={70}
            color={Colors.iconColorLight}
          />
        );
      case CellOccupancy.PLAYER_TWO:
        return <Ionicons name="close" size={76} color={Colors.iconColorDark} />;
      case CellOccupancy.NONE:
      default:
        return null;
    }
  };
  return (
    <Pressable {...props}>
      <View style={[styles.container, wrapperStyle]}>{getIcon(player)}</View>
    </Pressable>
  );
};

export default Cell;
