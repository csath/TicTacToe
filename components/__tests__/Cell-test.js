import * as React from 'react';
import renderer from 'react-test-renderer';

import Cell from '../Cell';

it(`Cell renders correctly`, () => {
  const tree = renderer.create(<Cell text='test'/>).toJSON();

  expect(tree).toMatchSnapshot();
  expect(tree.children.length).toBe(1);
});
