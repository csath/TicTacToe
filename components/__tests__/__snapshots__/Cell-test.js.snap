// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  text="test"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": 208.33333333333334,
          "justifyContent": "center",
          "width": 208.33333333333334,
        },
        undefined,
      ]
    }
  />
</View>
`;
